py/desispec/desi_proc_dashboard.py

```diff
@@ -0,0 +1,383 @@
+        cmd="chmod -R a+xr "+self.output_dir
+        os.system(cmd)
+
+    def _init_parser(self,parser):
+        parser.add_argument('-n','--nights', type=str, default = None, required = False, help="nights to monitor")
+        parser.add_argument('-nn','--n_night', type=str, default = None, required = False, help="all:all nights. ifdigit: the last n nights.")
+        parser.add_argument('-pd','--prod_dir', type=str, default = None, required = True, help="product base directory")
+        parser.add_argument('-od','--output_dir', type=str, default = None, required = True, help="output portal directory for the html pages ")
+        parser.add_argument('-ou','--output_url', type=str, default = None, required = True, help="output portal directory url ")
+        return parser
+
+    def calculate_one_night(self,night):
+        #print('{} Checking for new files on {}'.format(time.asctime(), night))
+        fileglob = '{}/{}/*/desi-*.fits.fz'.format(os.getenv('DESI_SPECTRO_DATA'), night)
+        known_exposures = set()
+        newexp = self.find_newexp(night, fileglob, known_exposures)
+        expids=[t[1] for t in newexp]
+        expids.sort(reverse=True)
+        output={}
+        for expid in expids:
+            # Check the redux folder for reduced files 
+            filename=os.getenv('DESI_SPECTRO_DATA')+'/'+str(night)+'/'+str(expid).zfill(8)+'/desi-'+str(expid).zfill(8)+'.fits.fz'
+            #h1=fitsio.read_header(filename,1)
+            h1=fits.getheader(filename,1)
+            flavor=h1['flavor'].strip()
+            #import pdb;pdb.set_trace()
+            try:
+                a= h1['SPCGRPHS']
+                b=a.split(',')
+            except:
+                a='SPX'
+                b=['SPX']
+            n_spectrographs=len(b)
+
+            exptime=str(h1['EXPTIME']).strip()
+            try:
+                obstype=h1['OBSTYPE'].strip()
+            except:
+                obstype='None' #h2=fitsio.read_header(filename,2)
+            #print(expid,flavor,obstype)
+            fileglob_psf='/project/projectdirs/desi/spectro/redux/daily/exposures/'+str(night)+'/'+str(expid).zfill(8)+'/psf*.fits'
+            fileglob_fit_psf='/project/projectdirs/desi/spectro/redux/daily/exposures/'+str(night)+'/'+str(expid).zfill(8)+'/fit-psf*.fits'
+            fileglob_fiberflat='/project/projectdirs/desi/spectro/redux/daily/exposures/'+str(night)+'/'+str(expid).zfill(8)+'/fiberflat*.fits'
+            fileglob_frame='/project/projectdirs/desi/spectro/redux/daily/exposures/'+str(night)+'/'+str(expid).zfill(8)+'/frame*.fits'
+            fileglob_sframe='/project/projectdirs/desi/spectro/redux/daily/exposures/'+str(night)+'/'+str(expid).zfill(8)+'/sframe*.fits'
+            fileglob_cframe='/project/projectdirs/desi/spectro/redux/daily/exposures/'+str(night)+'/'+str(expid).zfill(8)+'/cframe*.fits'
+            fileglob_sky='/project/projectdirs/desi/spectro/redux/daily/exposures/'+str(night)+'/'+str(expid).zfill(8)+'/sky*.fits'
```

Comment:
Please use the script options to generate these paths, e.g. `--prod_dir`, with a default of `prod_dir = desispec.io.specproc_root()`.

In the end, the code should not refer to "project" or "cdirs" anywhere, and preferably not `$DESI_SPECTRO_REDUX` either (using `specprod_root` and `rawdata_root` instead).